### PR TITLE
[PT-D] Enable nan_to_num op for sharded tensor

### PR DESCRIPTION
**Pull request by fduwjj**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #78223

Differential Revision: [D36649271](https://our.internmc.facebook.com/intern/diff/D36649271)

### CODE REVIEW

**Comment by fduwjj** [comment]
@fduwjj has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36649271).

File: torch/distributed/_shard/sharding_spec/chunk_sharding_spec_ops/matrix_ops.py [hunk 1]
```diff
@@ -418,3 +418,8 @@ def sharded_layer_norm(args, kwargs, pg):
     extra_check=sharded_layer_norm_check,
     customized_func=sharded_layer_norm,
 )
+
+
+_register_sharded_op_on_local_tensor(
+    torch.nan_to_num,
+)
```
Comment (pritamdamania87) [inline torch/distributed/_shard/sharding_spec/chunk_sharding_spec_ops/matrix_ops.py:425]: Isn't this an elementwise op instead of a matrix_op?

**Review by pritamdamania87 — APPROVED** [review]

File: torch/distributed/_shard/sharding_spec/chunk_sharding_spec_ops/matrix_ops.py (hunk 1 above)
Comment (fduwjj) [reply torch/distributed/_shard/sharding_spec/chunk_sharding_spec_ops/matrix_ops.py:425]: Right. Let me move it here.

**Comment by fduwjj** [comment]
@fduwjj has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36649271).

**Pull request closed without merging**